packages/plugin-core/src/web/engine/IReducedEngineApiService.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  BulkResp,
+  BulkWriteNotesOpts,
+  EngineDeleteNoteResp,
+  EngineDeleteOpts,
+  EngineWriteOptsV2,
+  FindNoteOpts,
+  NoteChangeEntry,
+  NoteProps,
+  NotePropsMeta,
+  QueryNotesOpts,
+  RenameNoteOpts,
+  RenameNotePayload,
+  RespV2,
+} from "@dendronhq/common-all";
+
+// Subset of IEngineAPIService
+export interface IReducedEngineAPIService {
+  /**
+   * Get NoteProps by id. If note doesn't exist, return undefined
+   */
+  getNote: (id: string) => Promise<NoteProps | undefined>;
```

Comment:
Just heads up that I changed the signature from https://github.com/dendronhq/dendron/pull/3359. I feel it'll be useful to propagate any errors to the clients